### PR TITLE
Refactor OpenApiTypesense.Connection.new/1

### PR DESCRIPTION
**Pull request by kianmeng**
Adding missing validation and tests when making a new connection.

## Summary by Sourcery

Refactor the Connection.new/1 function to include validation for required fields and add corresponding tests to ensure proper functionality and error handling.

Enhancements:
- Add validation for required fields in the Connection.new/1 function to ensure all necessary fields are provided.

Tests:
- Introduce new tests for the Connection.new/1 function to verify correct struct creation and error handling for missing or invalid data.

### CODE REVIEW

File: lib/open_api_typesense/connection.ex [hunk 1]
```diff
@@ -20,33 +20,52 @@ defmodule OpenApiTypesense.Connection do
...
         scheme: "http",
         ...
       }
+
+      iex> Connection.new(%{})
+      ** (ArgumentError) Missing required fields: [:port, :scheme, :host, :api_key]
+          (open_api_typesense 0.2.0) lib/open_api_typesense/connection.ex:56: OpenApiTypesense.Connection.new/1
+      iex:2: (file)
+
   """
```
Comment (kianmeng) [inline lib/open_api_typesense/connection.ex:44]: For readability, it's best to add empty line after the `## Examples` and before the closing `"""` heredocs. Not sure if you want to adopt this to all comments in the project.

File: lib/open_api_typesense/connection.ex [hunk 2]
```diff
@@ -20,33 +20,52 @@ defmodule OpenApiTypesense.Connection do
...
+    raise ArgumentError, "Expected a map for connection options"
+  end
+
+  @spec required_fields :: map()
+  defp required_fields do
+    struct(__MODULE__, %{}) |> Map.drop([:__struct__]) |> Map.keys()
   end
 
-  @doc since: "0.2.0"
   @spec defaults :: map()
```
Comment (kianmeng) [inline lib/open_api_typesense/connection.ex:69]: ExDoc does not render doc for private function. Do you still want to keep the `@doc` tag?

File: test/open_api_typesense_connection_test.exs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  test "new/1 with empty map raises ArgumentError" do
+    msg = "Missing required fields: [:port, :scheme, :host, :api_key]"
+    assert_raise ArgumentError, msg, fn -> Connection.new(%{}) end
+  end
+
+  test "new/1 with invalid data type raises ArgumentError" do
+    assert_raise ArgumentError, fn -> Connection.new("invalid") end
+  end
+end
```
Comment (kianmeng) [inline test/open_api_typesense_connection_test.exs:42]:
```
$ mix test --cover
Running ExUnit with seed: 952147, max_cases: 24

......................................................................
Finished in 1.6 seconds (1.5s async, 0.08s sync)
70 tests, 0 failures
----------------
COV    FILE                                        LINES RELEVANT   MISSED
 88.5% lib/open_api_typesense/client.ex              197       35        4
100.0% lib/open_api_typesense/connection.ex           78        8        0
100.0% lib/open_api_typesense/operations/analyt      213       21        0
100.0% lib/open_api_typesense/operations/collec      297       33        0
100.0% lib/open_api_typesense/operations/conver      168       18        0
100.0% lib/open_api_typesense/operations/curati      115       13        0
100.0% lib/open_api_typesense/operations/debug.       43        4        0
100.0% lib/open_api_typesense/operations/docume      511       66        0
100.0% lib/open_api_typesense/operations/health       33        3        0
100.0% lib/open_api_typesense/operations/keys.e      119       14        0
100.0% lib/open_api_typesense/operations/operat      110       12        0
100.0% lib/open_api_typesense/operations/overri       38        4        0
100.0% lib/open_api_typesense/operations/preset      135       15        0
100.0% lib/open_api_typesense/operations/stopwo      147       16        0
100.0% lib/open_api_typesense/operations/synony      138       16        0
[TOTAL]  98.5%
----------------
```

Probably is a good idea to add tests to `lib/open_api_typesense/client.ex`? Good jobs on adding moar tests to it!

**Review by jaeyson — APPROVED** [review]

**Comment by jaeyson** [comment]
Hi @kianmeng, thanks for the PR. There's an error in CI test: 

https://github.com/jaeyson/open_api_typesense/actions/runs/12332042761/job/34423378550?pr=2

```bash
  1) test new/1 with empty map raises ArgumentError (OpenApiTypesense.ConnectionTest)
Error:      test/open_api_typesense_connection_test.exs:34
     Wrong message for ArgumentError
     expected:
       "Missing required fields: [:port, :scheme, :host, :api_key]"
     actual:
       "Missing required fields: [:api_key, :host, :port, :scheme]"
     code: assert_raise ArgumentError, msg, fn -> Connection.new(%{}) end
     stacktrace:
       test/open_api_typesense_connection_test.exs:36: (test)

..................................................................
Finished in 0.6 seconds (0.6s async, 0.01s sync)
70 tests, 1 failure
```

Is it the order of the list `[:api_key, :host, :port, :scheme]`?

**Comment by kianmeng** [comment]
:partying_face: :partying_face: :partying_face: :partying_face: :partying_face:

**Pull request merged**